src/enclave/enclave.h

```diff
@@ -86,10 +90,22 @@ namespace enclave
       logger::config::msg() = AdminMessage::log_msg;
       logger::config::writer() = writer_factory.create_writer_to_outside();
 
+      // From
+      // https://software.intel.com/content/www/us/en/develop/articles/how-to-use-the-rdrand-engine-in-openssl-for-random-number-generation.html
+      if (
+        ENGINE_load_rdrand() != 1 ||
+        (rdrand_engine = ENGINE_by_id("rdrand")) == nullptr ||
+        ENGINE_init(rdrand_engine) != 1 ||
+        ENGINE_set_default(rdrand_engine, ENGINE_METHOD_RAND) != 1)
```

Comment:
If ENGINE_set_default() fails, are we leaking rdrand_engine?

Reply:
Yes, but everything will be aborted anyways and there wasn't an existing error-path. I can add that though.